proof-producer/CMakeLists.txt

```diff
@@ -20,6 +20,16 @@ include(CMConfig)
 include(CMDeploy)
 include(CMSetupVersion)
 
+option(PROOF_PRODUCER_STATIC_BINARIES "Link proof-producer binaries statically" OFF)
+
+if(PROOF_PRODUCER_STATIC_BINARIES)
+    set(BUILD_SHARED_LIBS OFF)
+    set(Boost_USE_STATIC_LIBS ON)
+else()
+    set(BUILD_SHARED_LIBS OFF)  # TODO: fix issue with precompiled headers to build shared libraries
```

Comment:
Is it possible to refer to the issue somehow (or at least briefly describe it)?

Reply:
I was trying to revive build with shared libraries (they were abandoned recently), but got this error:
```
       > cc1plus: error: /build/source/build/proof-producer/libs/output_artifacts/CMakeFiles/proof_generatorOutputArtifacts.dir/cmake_pch.hxx.gch: not used because `proof_generatorOutputArtifacts_EXPORTS' not defined [-Werror=invalid-pch]
       > cc1plus: all warnings being treated as errors
       > ninja: build stopped: subcommand failed.
```
To be honest, I still don't know what went wrong, and since this is not a subject of this PR, I decided to fix it later

Reply:
We can don't use shared libraries at all

Reply:
Added error code to comment